.github/workflows/ci.yml

```
@@ -31,6 +31,18 @@ jobs:
           override: true
       - run: cargo build --target ${{ matrix.target }} --release --no-default-features
 
+  # Mimics the setup of docs.rs, but fails on warnings
+  build-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          override: true
+          profile: minimal
+      - run: env RUSTDOCFLAGS='--cfg docsrs -D warnings' cargo doc --all-features
```

```suggestion
      - run: env RUSTDOCFLAGS='--cfg docsrs -D warnings' cargo doc --all-features --no-deps
```
…to avoid documenting dependencies.

Also, why not use an `env` key for the flags? Maybe like so:

```
build-docs:
   runs-on: ubuntu-latest
   env:
       RUSTDOCFLAGS: -D warnings
```

Also-2: What does `--cfg docsrs` do? Is it an undocumented feature?

---

The all-deps is necessary because this way an invalid link to a dependency (or standard library) item will be detected. 

The env thing is just a stylistic choice really, it's just one command, and if there's a fail in CI there someone can just copy and paste it and run it locally.